pkg/agent/manager/cache/cache.go

```diff
@@ -310,7 +310,12 @@ func (c *Cache) UpdateEntries(update *UpdateEntries, checkSVID func(*common.Regi
 		federatedBundlesChanged := len(fedAdd) > 0 || len(fedRem) > 0
 		if !federatedBundlesChanged {
 			for _, id := range newEntry.FederatesWith {
-				if bundleChanged[id] {
+				td, err := spiffeid.TrustDomainFromString(id)
+				if err != nil {
+					c.log.WithField(telemetry.TrustDomainID, id).Warn("Federated Trust Domain invalid")
```

Comment:
We should add the error as a field
```suggestion
					c.log.WithFields(logrus.Fields{
					    telemetry.TrustDomainID: id,
					    logrus.ErrorKey: err,
					}).Warn("Invalid federated trust domain")
```